app/models/miq_worker/runner.rb

```diff
@@ -374,6 +375,32 @@ def heartbeat_to_drb
   def heartbeat_to_file(timeout = nil)
     timeout ||= worker_settings[:heartbeat_timeout] || Workers::MiqDefaults.heartbeat_timeout
     File.write(@worker.heartbeat_file, (Time.now.utc + timeout).to_s)
+
+    get_messages.each { |msg, *args| process_message(msg, *args) }
+  end
+
+  def get_messages
+    messages = []
+    now = Time.now.utc
+    @my_last_config_change ||= now
+
+    last_config_change = server_last_change(:last_config_change)
+    if last_config_change && last_config_change > @my_last_config_change
+      _log.info("#{log_prefix} Configuration has changed, New TS: #{last_config_change}, Old TS: #{@my_last_config_change}")
+      messages << ["sync_config"]
+
+      @my_last_config_change = last_config_change
+    end
+
+    messages
+  end
+
+  def key_store
+    @key_store ||= Dalli::Client.new(MiqMemcached.server_address, :namespace => "server_monitor")
+  end
```

Comment:
Good point.  Is it premature?  My hope was that we'd know better about how this would look when we figure out how and if we'd use memcached/redis/etc. to tell a worker (who's caught a SIGINT/SIGTERM and continued finishing the queue work it was doing) to exit.  Either way, @NickLaMuro's research into how we let workers gracefully finish their work might drive the interface for this.